src/main/java/nl/hsac/fitnesse/junit/HsacFitNesseRunner.java

```diff
@@ -136,6 +140,33 @@ protected void runPages(List<WikiPage> pages, RunNotifier notifier) {
 
     }
 
+    // In the original runner class, we cannot use a system property as annotation value,
+    // so we override it to work exactly like the suiteFilter annotation.
+    // This also needs the ExcludeSuiteFilter interface
+    @Override
+    protected String getExcludeSuiteFilter(Class<?> klass) throws Exception {
+        ExcludeSuiteFilter excludeSuiteFilterAnnotation = (ExcludeSuiteFilter)klass.getAnnotation(ExcludeSuiteFilter.class);
+        if (excludeSuiteFilterAnnotation == null) {
```

Comment:
Currently `HsacFitNesseRunner` makes the system property `fitnesseSuiteToRun` 'win' from the string literal in the `Suite` annotation (even if the Java test class code does not reference the system property). This allows the developer to commit any value in the test class without breaking the build server's run (which sets the system property).

The approach for `(Exclude)SuiteFilter` in this version is that the 'value' (literal `value` in Java code) wins from the system property.
Can we change the behavior for `(Exclude)SuiteFilter` so that the system properties (with names you use as examples) always win from the annotation value in code (even let the filters become active is the annotation was not present on the Java class)?
I believe all it would take is an extra (first) `if`-branch, turning the current first into an `else if`, checking whether the system property has a value and using that value if so.
Of course this does also mean we have to override the current behavior for the `SuiteFilter` annotation.

I also agree that the current code would be a good improvement for the FitNesse project's `FitNesseRunner`. Are you planning to send them a pull request with it?